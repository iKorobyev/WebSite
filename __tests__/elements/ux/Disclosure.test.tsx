import React from 'react'
import { cleanup, render, waitFor, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Disclosure } from '~ux'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Disclosure', () => {
  it('renders correctly', async () => {
    const props = {
      title: 'Test',
      description: 'Test'
    }

    const { container } = render(<Disclosure {...props} />)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
  })
})
