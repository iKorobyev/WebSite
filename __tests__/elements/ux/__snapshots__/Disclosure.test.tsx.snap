// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclosure renders correctly 1`] = `
<div>
  <div
    class=""
  >
    <button
      class="flex justify-between w-full px-4 py-2 text-sm font-medium text-left text-green-900 bg-green-100 rounded-lg hover:bg-green-200 focus:outline-none focus-visible:ring focus-visible:ring-green-500 focus-visible:ring-opacity-75"
      id="headlessui-disclosure-button-1"
      type="button"
    >
      <span>
        Test
      </span>
      <svg
        class="w-20 h-20 text-green-500"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M14.707 12.707a1 1 0 01-1.414 0L10 9.414l-3.293 3.293a1 1 0 01-1.414-1.414l4-4a1 1 0 011.414 0l4 4a1 1 0 010 1.414z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;
